Model/OrderData.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Metrilo\Analytics\Model;
+
+class OrderData
+{
+    public function __construct(
+        \Magento\Sales\Model\ResourceModel\Order\CollectionFactory $orderCollection
+    ) {
+        $this->orderCollection  = $orderCollection;
+    }
+
+    public function getOrders($storeId)
+    {
+        $ordersArray   = [];
+        $orders        = $this->getOrderQuery($storeId);
+
+        foreach ($orders as $order) {
+            if(!trim($order->getCustomerEmail())) {
+                continue;
+            }
+            $orderItems = $order->getAllItems();
+            $orderProducts = [];
+            
+            foreach ($orderItems as $orderItem) {
+                $itemType = $orderItem->getProductType();
+                if ($itemType == 'configurable' || $itemType == 'bundle') { // exclude configurable/bundle parent product returned by getAllItems() method
+                    continue;
+                }
+                $orderProducts[] = [
+                    'productId' => $orderItem->getProductId(),
+                    'quantity'  => $orderItem->getQtyOrdered()
+                ];
+            }
+
+            $orderBillingData = $order->getBillingAddress();
+            $street           = $orderBillingData->getStreet();
+            $couponCode       = [];
+            if ($order->getCouponCode()) {
+                $couponCode[] = $order->getCouponCode();
+            }
+
+            $orderBilling = [
+                "firstName"     => $orderBillingData->getFirstname(),
+                "lastName"      => $orderBillingData->getLastname(),
+                "address"       => is_array($street) ? implode(PHP_EOL, $street) : $street,
+                "city"          => $orderBillingData->getCity(),
+                "country"       => $orderBillingData->getCountryId(),
+                "phone"         => $orderBillingData->getTelephone(),
+                "postcode"      => $orderBillingData->getPostcode(),
+                "paymentMethod" => $order->getPayment()->getMethodInstance()->getTitle()
+            ];
+
+            $ordersArray[] = [
+                'id'        => $order->getIncrementId(),
+                'createdAt' => strtotime($order->getCreatedAt()),
+                'updatedAt' => strtotime($order->getUpdatedAt()),
+                'email'     => $order->getCustomerEmail(),
+                'amount'    => $order->getBaseGrandTotal(),
+                'coupons'   => $couponCode, // RETURNING NULL if there is no coupon applied // MULTYPLE coupon codes are available via extension ONLY
+                'status'    => $order->getStatus(),
+                'products'  => $orderProducts,
+                'billing'   => $orderBilling
+            ];
+
+            unset($couponCode);
```

Comment:
why do you have to do this?

Reply:
After the refactoring of the way couponCode variable is constructed the unsetting of the variable is no longer required.